.github/workflows/ci.yml

```diff
@@ -45,7 +45,7 @@ jobs:
 
     - name: Build OpenSearch
       working-directory: ./OpenSearch
-      run: ./gradlew publishToMavenLocal -Dbuild.snapshot=false
+      run: ./gradlew publishToMavenLocal -Dbuild.version_qualifier=beta1 -Dbuild.snapshot=false
```

Comment:
`-rc1`

Reply:
@dblock Both `rc1` and removing `-Dbuild.version_qualifier` fails security plugin build, which is this [issue](https://github.com/opensearch-project/OpenSearch/issues/762) for.

Reply:
The problem was introduced in https://github.com/opensearch-project/security/pull/1174 that removes qualifier but does not update pom dependency. I wonder how that PR was tested and approved.

Reply:
This works here because you have hard-coded beta1 in the plugin for the OpenSearch dependency. See https://github.com/opensearch-project/OpenSearch/issues/762#issuecomment-846291955. 

Change this to be rc1, and update your code to say rc1 instead beta1. 

These qualifiers don't do anything in the build as long as they are consistent so that the dependency on OpenSearch can be resolved, but in order for the plugin to start the version of the engine and the plugin will have to match, and the next version is rc1.

Reply:
> This works here because you have hardcoded beta1 in pom.xml for the OpenSearch dependency. See opensearch-project/OpenSearch#762 (comment).

@dblock I guess you refer to CI, but it works only because maven artifacts are cached: https://github.com/opensearch-project/security/blob/5a81a424f4dc0dd1e72df9d8c27710e9dbcdec1b/.github/workflows/ci.yml#L32 Without caching maven artifacts or on a fresh box with no prior build of beta1 artifacts, the build would fail.

> These qualifiers don't do anything in the build as long as they are consistent so that the dependency on OpenSearch can be resolved, but in order for the plugin to start the version of the engine and the plugin will have to match, and the next version is rc1.

If `rc1` does not correspond to a commit/tag in Opensearch, why is it used? If plugins are supposed to be built against the latest commit of 1.x branch, it should use `SNAPSHOT` (this is the meaning of `SNAPSHOT`), not an `rc1` tag/qualifier.

Reply:
Yes re: cached maven artifacts, I didn't see that, should probably be removed now

You're right on rc1. If we had Maven we would be able to say exactly what you're building against, which could be an rc1 eventually. Right now if you called everything -SNAPSHOT you wouldn't have a way to rename that to rc1 because it would need a change in code. I think we need to fix https://github.com/opensearch-project/OpenSearch/issues/763 first, then we can undo this craziness. It's just a workaround to produce something called `-rc1` everywhere.

Reply:
> Yes re: cached maven artifacts, I didn't see that, should probably be removed now

Caching maven artifacts should not be removed, it works as designed to speed up security plugin build and avoid loading artifacts from maven on each build. What should be removed is using CI to build and publish Opensearch artifacts to local maven repo.

Reply:
You can be right, or you can prevent a similar problem from happening in the short term. Your call.

Reply:
Similar problems should be prevented by testing and PR review, not by removing what worked before and is not broken. Even if caching is removed, what prevents CI of using wrong branch/tag and naming it `beta1` or `rc1`?